front/components/assistant/details/AssistantActionsSection.tsx

```diff
@@ -0,0 +1,423 @@
+import {
+  BracesIcon,
+  CommandLineIcon,
+  ExternalLinkIcon,
+  FolderIcon,
+  Icon,
+  IconButton,
+  PlanetIcon,
+  Tree,
+} from "@dust-tt/sparkle";
+import type {
+  AgentActionConfigurationType,
+  AgentConfigurationType,
+  ContentNodesViewType,
+  DataSourceConfiguration,
+  DataSourceViewType,
+  DustAppRunConfigurationType,
+  LightWorkspaceType,
+  RetrievalConfigurationType,
+  TablesQueryConfigurationType,
+} from "@dust-tt/types";
+import {
+  assertNever,
+  isBrowseConfiguration,
+  isDustAppRunConfiguration,
+  isProcessConfiguration,
+  isRetrievalConfiguration,
+  isTablesQueryConfiguration,
+  isWebsearchConfiguration,
+} from "@dust-tt/types";
+import { useMemo, useState } from "react";
+
+import DataSourceViewDocumentModal from "@app/components/DataSourceViewDocumentModal";
+import { DataSourceViewPermissionTree } from "@app/components/DataSourceViewPermissionTree";
+import { getContentNodeInternalIdFromTableId } from "@app/lib/api/content_nodes";
+import { GLOBAL_AGENTS_SID } from "@app/lib/assistant";
+import { getConnectorProviderLogoWithFallback } from "@app/lib/connector_providers";
+import { getVisualForContentNode } from "@app/lib/content_nodes";
+import {
+  canBeExpanded,
+  getDisplayNameForDataSource,
+  isFolder,
+} from "@app/lib/data_sources";
+import {
+  useDataSourceViewContentNodes,
+  useDataSourceViews,
+} from "@app/lib/swr/data_source_views";
+import { classNames } from "@app/lib/utils";
+
+interface AssistantActionsSectionProps {
+  agentConfiguration: AgentConfigurationType;
+  owner: LightWorkspaceType;
+}
+
+export function AssistantActionsSection({
+  agentConfiguration,
+  owner,
+}: AssistantActionsSectionProps) {
+  const isDustGlobalAgent = agentConfiguration.sId === GLOBAL_AGENTS_SID.DUST;
```

Comment:
`nit`: since it's only used within the memo, could we move it inside the `useMemo`?

Reply:
Good point.